components/Buttons/src/Theming/MDCButton+MaterialTheming.m

```diff
@@ -144,11 +145,22 @@ - (void)applyTextThemeWithColorScheme:(id<MDCColorScheming>)colorScheme {
 }
 
 - (void)applyTextThemeWithTypographyScheme:(id<MDCTypographyScheming>)typographyScheme {
-  [MDCButtonTypographyThemer applyTypographyScheme:typographyScheme toButton:self];
+  [self resetTitleForAllStates];
+  [self setTitleFont:typographyScheme.button forState:UIControlStateNormal];
 }
 
 - (void)applyTextThemeWithShapeScheme:(id<MDCShapeScheming>)shapeScheme {
   [MDCButtonShapeThemer applyShapeScheme:shapeScheme toButton:self];
 }
 
+#pragma mark - General helpers
+
+- (void)resetTitleForAllStates {
```

Comment:
Maybe this should be called `resetTitleFontForAllStates` too. It's kind of a bummer we have these two methods doing the same thing :/

Reply:
Yeah that's unfortunate, I removed this method as it's only used once and I don't know how it would play with it's `super` having the same method but in an extension.